### PR TITLE
Converge cloudserver images

### PR DESCRIPTION
**Pull request by francoisferrand**
Build federation image as additional layers on top of "standard" cloudserver image, to avoid duplication and ensure the image we tested is used in production.
There are still some differences at the moment (e.g. name of user, path) which should be further converged to minimize differences, may be done over time after considering all things.

This allows us to fully control (and minimize) what gets into the image, relying on any official node image. Also, now switch from python-based supervisord to a golang reimplementation [1], to minimize dependencies and CVEs.

Finally, took the change to fix startup command of the image, to ensure it properly propagates OS signals.

[1] ochinchina/supervisord

Issue: CLDSRV-597

### CODE REVIEW

File: images/federation/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
+ARG BACKBEAT_VERSION=latest
```
Comment (benzekrimaha) [inline images/federation/Dockerfile:1]: ARG CLOUDSERVER_VERSION=latest

Comment (benzekrimaha) [inline images/federation/Dockerfile]: `ARG CLOUDSERVER_VERSION=latest`

File: images/federation/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+RUN mkdir ${LOG_DIR} && chown ${USER} ${LOG_DIR} && \
+    mkdir ${CONF_DIR} && chown ${USER} ${CONF_DIR} && \
+    mkdir ${DATA_DIR} && chown ${USER} ${DATA_DIR} && \
+    mkdir -m 777 ${SUP_RUN_DIR} && chown ${USER} ${SUP_RUN_DIR}
+
+USER ${USER}
+WORKDIR ${HOME_DIR}/s3
+
+# Keep same output as chown command without group (use group 0)
+COPY --chown=${USER}:0 --from=builder /usr/src/app ${HOME_DIR}/backbeat
```
Comment (benzekrimaha) [inline images/federation/Dockerfile]: COPY --chown=${USER}:0 --from=builder /usr/src/app ${HOME_DIR}/cloudserver

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -177,6 +180,23 @@ jobs:
...
+      - name: Build and push federation image
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          context: images/federation
+          provenance: false
+          build-args:
+            CLOUDSERVER_VERSION=${{ github.sha }}
+          tags: |
+            ghcr.io/${{ github.repository }}:${{ github.sha }}-federation
```
Comment (ghost) [inline .github/workflows/tests.yaml:193]: shall we seize this opportunity to use a version before the sha?

Comment (francoisferrand) [reply .github/workflows/tests.yaml:193]: no completely trivial (need to compute the version) : will be done separately.
Also, this is needed on 8.8 as well: while this target 9.0 only.

**Review by ghost — APPROVED** [review]

**Review by benzekrimaha — APPROVED** [review]

**Review by rachedbenmustapha — COMMENTED** [review]
Would be good to post to the #eng-federation chan for review.

**Pull request merged**